pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -758,6 +776,11 @@ func (c *Controller) syncRule(key string) error {
 		return nil
 	}
 
+	var isNodeNetworkPolicy bool
+	if c.nodeNetworkPolicyEnabled {
+		isNodeNetworkPolicy = rule.isNodeNetworkPolicyRule()
```

Comment:
Good question.  How about this:

1. Skip reconciling the rule and log something like "Feature gate NodeNetworkPolicy is not enabled, skipping reconciling the rule.
2. Report the status. If so, I think we need to enhance the StatusManager methods like to report the status that the rule cannot synced like in PR #5591. Another way is that we don't report anything, but this seems not friendly to users since users can only find that the Node NetworkPolicies cannot be synced in antrea-agent log.

Reply:
I feel just ignoring Node policy when the feature is not enabled is enough. But logging it's skipped due to disabled feature gate is harmless anyway.